packages/core/src/components/carousel/carousel.tsx

```diff
@@ -0,0 +1,104 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Prop,
+  h,
+} from '@stencil/core'
+import 'swiper/swiper-element-bundle.js'
+
+import { AutoplayOptions, PaginationOptions, Swiper } from 'swiper/types'
+
+interface SwiperElement extends HTMLElement {
+  swiper: Swiper
+}
+@Component({
+  tag: 'atom-carousel',
+  styleUrl: 'carousel.scss',
+  shadow: false,
+})
+export class AtomCarousel {
+  @Element() host!: HTMLElement
+  private swiperEl: SwiperElement
+  @Event() atomClickPrev!: EventEmitter<string>
+  @Event() atomClickNext!: EventEmitter<string>
+  @Event() atomChange!: EventEmitter<string>
+
+  @Prop({ mutable: true })
+  autoplay?: boolean
+  @Prop({ mutable: true })
+  autoplayDelay?: AutoplayOptions['delay']
+  @Prop({ mutable: true })
+  freeMode?: boolean
+  @Prop({ mutable: true })
+  loop?: boolean
+  @Prop({ mutable: true })
+  navigation?: boolean = true
+  @Prop({ mutable: true })
+  pagination?: boolean = true
+  @Prop({ mutable: true })
+  paginationClickable?: boolean
+  @Prop({ mutable: true })
+  paginationType?: PaginationOptions['type'] = 'bullets'
+  @Prop({ mutable: true })
+  slidesPerGroup?: number | string = 1
+  @Prop({ mutable: true })
+  slidesPerView?: number | string = 1
+  @Prop({ mutable: true })
+  spaceBetween?: number = 0
+  @Prop({ mutable: true }) speed?: number
+
+  // injectStyles is needed in order to overwrite css by class inside shadow root with ::part
+  private injectStyles = ['.swiper-button-disabled { opacity: 0 !important}']
+
+  private convertChildren() {
```

Comment:
Question: Does make sense to add the ADR link here?

Reply:
@tassioFront about the stylesInject? I confess i havent saw this as an architectural desicion once this is the way swiper  docs indicate to overwrite or add styles to shadow DOM scope, [is in their docs](https://swiperjs.com/element#injecting-styles), was not our decision in fact.

But as we already have an ADR about carousel and the swiper i can add this point there as a caveat, what you think?

Reply:
> But as we already have an ADR about carousel and the swiper i can add this point there as a caveat, what you think?

I'm talking about this one, actually @schirrel  hahahaha Add de ADR link of Carousel and swiper at `convertChildren` function, thinking people might not understand this function at the first touch.

Reply:
Ooooh i got now 🤣  makes total sense, i think i forgot to add here and added only to carousel item 😅

Reply:
done [3983924](https://github.com/juntossomosmais/atomium/pull/235/commits/3983924369a620a702d271b83506631316ec4eff)

Reply:
hahahahahaha perfect, my friend